src/MergeContext.js

```diff
@@ -1166,6 +1237,8 @@ class PullRequest {
 
         assert(!this._stagingBanned);
 
+        await this._setAutomatedMergeStatusForStaged("success");
```

Comment:
Since we are setting this a bit prematurely/optimistically, it would be nice to make this one `success (will merge SHA)` and then, if the status may become visible anywhere after we successfully merge (as discussed in another change request), change that to `success (M-merged as SHA)`.